### PR TITLE
Reorganization

### PR DESCRIPTION
**Pull request by christhomson**
@RyanSeto thoughts?

I reorganized everything to separate concerns a bit. Before, `routes/room.coffee` had [a lot of random stuff in it](https://github.com/christhomson/roomular/blob/eb6916c3ab42cb4fa82b5ab15a8bb091bd1bbccb/routes/room.coffee), and it was hard to follow.

There are still bits here and there that could use some refactoring (like the way we figure out where gaps should go), but at least now things are sufficiently separate.

### CODE REVIEW

**Comment by RyanSeto** [comment]
Looks fine and dandy from an initial quick glance over it.

**Comment by christhomson** [comment]
One thing to note is this does not preserve the same URLs that worked before. But no one knows that Roomular exists so that shouldn't be an issue.

:ship:able?

**Comment by jasonchutko** [comment]
-2. We should maintain backwards compatibility. Please fix before :ship:ing.

**Comment by christhomson** [comment]
Done! :checkered_flag:

**Comment by RyanSeto** [comment]
:shipit:?!
